src/lib/util.js

```diff
@@ -41,8 +40,53 @@ function getRejectedPromiseIfTimedOut (timeoutMs) {
   );
 }
 
+/**
+ * Wraps an onSuccess and onError handler around a function
+ * @param {function} orignalFn
+ * @param {function} [onError]
+ * @param {function} [onSuccess]
+ * @return {*} - returns whatever the original function would have returned
+ */
+const wrapCompletedHandlers = (orignalFn, onError, onSuccess) => (...args) => {
+  let returnVal;
+  const callOnSuccess = (returnVal) => {
+    if (isFunction(onSuccess)) {
+      onSuccess(returnVal, args);
+    }
+  };
+  const callOnError = (error) => {
+    if (isFunction(onError)) {
+      onError(error);
+    }
+  };
+
+  try {
+    returnVal = orignalFn(...args);
```

Comment:
Is this a case for Promise.try?

Reply:
Promise.try is a bluebird function, which is only available in dev (for tests). Trying not to bring it in for the core package